src/integrationtests/java/com/aws/greengrass/integrationtests/ConfigTest.java

```diff
@@ -78,6 +93,92 @@ void GIVEN_Greengrass_with_mqtt_bridge_WHEN_multiple_config_changes_consecutivel
+
+        Topics config = testContext.getKernel().locate(MQTTBridge.SERVICE_NAME).getConfig()
+                .lookupTopics(CONFIGURATION_CONFIG_KEY).lookupTopics("brokerClient");
+
+        // publish a small message and verify that it is received
+        String topic = "topic/toLocal";
+        Set<String> topics = new HashSet<>();
+        topics.add(topic);
+
+        MqttMessage expectedMessage = MqttMessage.builder()
+                .topic(topic)
+                .payload("abc".getBytes(StandardCharsets.UTF_8))
+                .userProperties(Collections.singletonList(new UserProperty("key", "val")))
+                .responseTopic("response topic")
+                .messageExpiryIntervalSeconds(1234L)
+                .payloadFormat(Publish.PayloadFormatIndicator.UTF8)
+                .contentType("contentType")
+                .build();
+
+        Pair<CompletableFuture<Void>, Consumer<MqttMessage>> messageHandler =
+                asyncAssertOnConsumer(message -> assertEquals(Arrays.toString(expectedMessage.getPayload()),
+                        Arrays.toString(message.getPayload())));
+
+        testContext.getLocalV5Client().updateSubscriptions(topics, messageHandler.getRight());
+        testContext.getLocalV5Client().publish(
+                MqttMessage.builder()
+                        .topic(topic)
+                        .payload("abc".getBytes(StandardCharsets.UTF_8))
+                        .userProperties(Collections.singletonList(new UserProperty("key", "val")))
+                        .responseTopic("response topic")
+                        .messageExpiryIntervalSeconds(1234L)
+                        .payloadFormat(Publish.PayloadFormatIndicator.UTF8)
+                        .contentType("contentType")
+                        .build());
+        messageHandler.getLeft().get(RECEIVE_PUBLISH_SECONDS, TimeUnit.SECONDS);
+
+        // change config values
+        config.lookup(BridgeConfig.KEY_SESSION_EXPIRY_INTERVAL).withValue(1);
+        config.lookup(BridgeConfig.KEY_MAXIMUM_PACKET_SIZE).withValue(1);
+        config.lookup(BridgeConfig.KEY_RECEIVE_MAXIMUM).withValue(1);
+
+        // verify that config changes take effect
+        assertThat("session expiry interval config update",
+                () -> testContext.getLocalV5Client().getSessionExpiryInterval(), eventuallyEval(is(1L)));
+        assertThat("maximum packet size config update", () -> testContext.getLocalV5Client().getMaximumPacketSize(),
+                eventuallyEval(is(1L)));
+        assertThat("receive maximum config update", () -> testContext.getLocalV5Client().getReceiveMaximum(),
+                eventuallyEval(is(1)));
+
+        // Publish a large message to IoT Core and verify that it is not received due to the local client's config
+        // We expect a timeout since the previous message was too large to successfully publish, given the local
```

Comment:
this and the above feel like theyre saying the same thing

Reply:
missed that, updated the comments